sentry-android-core/src/main/java/io/sentry/android/core/SystemEventsBreadcrumbsIntegration.java

```diff
@@ -0,0 +1,197 @@
+import static android.content.Intent.ACTION_BATTERY_OKAY;
+import static android.content.Intent.ACTION_BOOT_COMPLETED;
+import static android.content.Intent.ACTION_BUG_REPORT;
+import static android.content.Intent.ACTION_CAMERA_BUTTON;
+import static android.content.Intent.ACTION_CONFIGURATION_CHANGED;
+import static android.content.Intent.ACTION_DATE_CHANGED;
+import static android.content.Intent.ACTION_DEVICE_STORAGE_LOW;
+import static android.content.Intent.ACTION_DEVICE_STORAGE_OK;
+import static android.content.Intent.ACTION_DOCK_EVENT;
+import static android.content.Intent.ACTION_INPUT_METHOD_CHANGED;
+import static android.content.Intent.ACTION_LOCALE_CHANGED;
+import static android.content.Intent.ACTION_MEDIA_BAD_REMOVAL;
+import static android.content.Intent.ACTION_MEDIA_MOUNTED;
+import static android.content.Intent.ACTION_MEDIA_UNMOUNTABLE;
+import static android.content.Intent.ACTION_MEDIA_UNMOUNTED;
+import static android.content.Intent.ACTION_POWER_CONNECTED;
+import static android.content.Intent.ACTION_POWER_DISCONNECTED;
+import static android.content.Intent.ACTION_REBOOT;
+import static android.content.Intent.ACTION_SCREEN_OFF;
+import static android.content.Intent.ACTION_SCREEN_ON;
+import static android.content.Intent.ACTION_SHUTDOWN;
+import static android.content.Intent.ACTION_TIMEZONE_CHANGED;
+import static android.content.Intent.ACTION_TIME_CHANGED;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import androidx.annotation.Nullable;
+import io.sentry.core.Breadcrumb;
+import io.sentry.core.IHub;
+import io.sentry.core.Integration;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.util.Objects;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.TestOnly;
+
+public final class SystemEventsBreadcrumbsIntegration implements Integration, Closeable {
+
+  private final @NotNull Context context;
+
+  @TestOnly @Nullable SystemEventsBroadcastReceiver receiver;
+
+  private @Nullable SentryAndroidOptions options;
+
+  private final @NotNull List<String> actions;
+
+  public SystemEventsBreadcrumbsIntegration(final @NotNull Context context) {
+    this(context, getDefaultActions());
+  }
+
+  public SystemEventsBreadcrumbsIntegration(
+      final @NotNull Context context, final @NotNull List<String> actions) {
+    this.context = Objects.requireNonNull(context, "Context is required");
+    this.actions = Objects.requireNonNull(actions, "Actions list is required");
+  }
+
+  @Override
+  public void register(final @NotNull IHub hub, final @NotNull SentryOptions options) {
+    Objects.requireNonNull(hub, "Hub is required");
+    this.options =
```

Comment:
that's a good idea, but I've done something similar already for integrations and the whole PR got inferred because of other reasons, this part specifically might be worth.